Seata AopProxyException: Could not generate CGLIB subclass of class [class com.sun.proxy.$Proxy21]

### Ⅰ. Issue Description
When i try to add a new  HSF Parser, the HSF bean have a own proxy type but not SpringProxy. Then throw org.springframework.aop.framework.AopConfigException

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Caused by: java.lang.RuntimeException: org.springframework.aop.framework.AopConfigException: Could not generate CGLIB subclass of class [class com.sun.proxy.$Proxy21]: Common causes of this problem include using a final class or a non-visible class; nested exception is java.lang.IllegalArgumentException: Cannot subclass final class class com.sun.proxy.$Proxy21
	at io.seata.spring.annotation.GlobalTransactionScanner.wrapIfNecessary(GlobalTransactionScanner.java:249)
	at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.postProcessAfterInitialization(AbstractAutoProxyCreator.java:325)
	at io.seata.spring.annotation.GlobalTransactionScanner.postProcessAfterInitialization(GlobalTransactionScanner.java:337)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsAfterInitialization(AbstractAutowireCapableBeanFactory.java:361)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.postProcessObjectFromFactoryBean(AbstractAutowireCapableBeanFactory.java:1429)
	at org.springframework.beans.factory.support.FactoryBeanRegistrySupport$1.run(FactoryBeanRegistrySupport.java:139)
	... 33 more
Caused by: org.springframework.aop.framework.AopConfigException: Could not generate CGLIB subclass of class [class com.sun.proxy.$Proxy21]: Common causes of this problem include using a final class or a non-visible class; nested exception is java.lang.IllegalArgumentException: Cannot subclass final class class com.sun.proxy.$Proxy21
	at org.springframework.aop.framework.Cglib2AopProxy.getProxy(Cglib2AopProxy.java:213)
	at org.springframework.aop.framework.ProxyFactory.getProxy(ProxyFactory.java:110)
	at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.createProxy(AbstractAutoProxyCreator.java:490)
	at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.wrapIfNecessary(AbstractAutoProxyCreator.java:365)
	at io.seata.spring.annotation.GlobalTransactionScanner.wrapIfNecessary(GlobalTransactionScanner.java:237)
	... 38 more
Caused by: java.lang.IllegalArgumentException: Cannot subclass final class class com.sun.proxy.$Proxy21
	at net.sf.cglib.proxy.Enhancer.generateClass(Enhancer.java:446)
	at net.sf.cglib.transform.TransformingClassGenerator.generateClass(TransformingClassGenerator.java:33)
	at net.sf.cglib.core.DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:25)
	at net.sf.cglib.core.AbstractClassGenerator.create(AbstractClassGenerator.java:216)
	at net.sf.cglib.proxy.Enhancer.createHelper(Enhancer.java:377)
	at net.sf.cglib.proxy.Enhancer.create(Enhancer.java:285)
	at org.springframework.aop.framework.Cglib2AopProxy.getProxy(Cglib2AopProxy.java:201)
	... 42 more
```


### Ⅲ. Describe what you expected to happen
The proposed solution is to wrap ‘AopUtils.isAopProxy(bean)’  make it return true, when use any proxy type of user define.
io.seata.spring.annotation.GlobalTransactionScanner.wrapIfNecessary(GlobalTransactionScanner.java:237)



                if (!AopUtils.isAopProxy(bean)) {
                    bean = super.wrapIfNecessary(bean, beanName, cacheKey);
                } else {
                    AdvisedSupport advised = SpringProxyUtils.getAdvisedSupport(bean);
                    Advisor[] advisor = buildAdvisors(beanName, getAdvicesAndAdvisorsForBean(null, null, null));
                    for (Advisor avr : advisor) {
                        advised.addAdvisor(0, avr);
                    }
                }


### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1.  Define a bean <hsf:consumer id="actionOne" interface="io.seata.edas.tcc.action.ActionOne"  version="1.0.0" />
2. Reference bean ‘actionOne’ 
3. Start your project

### Ⅴ. Anything else we need to know?


### Ⅵ. Environment:

- JDK version : java 8
- OS : Windows 7
- Others:

--- a/tcc/src/main/java/io/seata/rm/tcc/remoting/parser/HSFRemotingParser.java
+++ b/tcc/src/main/java/io/seata/rm/tcc/remoting/parser/HSFRemotingParser.java
@@ -1,0 +1,119 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.rm.tcc.remoting.parser;
+
+import io.seata.common.exception.FrameworkException;
+import io.seata.common.util.ReflectionUtil;
+import io.seata.rm.tcc.remoting.Protocols;
+import io.seata.rm.tcc.remoting.RemotingDesc;
+
+/**
+ * HSF Remote Bean Parser
+ *
+ * @author ppf@jiumao.org
+ */
+public class HSFRemotingParser extends AbstractedRemotingParser {
+
+    /**
+     * is HSF env
+     */
+    private static volatile boolean isHsf;
+
+    static {
+        // check HSF runtime env
+        try {
+            Class.forName("com.taobao.hsf.app.api.util.HSFApiConsumerBean");
+            Class.forName("com.taobao.hsf.app.api.util.HSFApiProviderBean");
+
+            Class.forName("com.taobao.hsf.app.spring.util.HSFSpringConsumerBean");
+            Class.forName("com.taobao.hsf.app.spring.util.HSFSpringProviderBean");
+
+            isHsf = true;
+        } catch (ClassNotFoundException e) {
+            isHsf = false;
+        }
+    }
+
+    @Override
+    public boolean isRemoting(Object bean, String beanName) {
+        return isHsf && (isReference(bean, beanName) || isService(bean, beanName));
+    }
+
+    @Override
+    public boolean isReference(Object bean, String beanName) {
+        String beanClassName = bean.getClass().getName();
+        return isHsf && ("com.taobao.hsf.app.spring.util.HSFSpringConsumerBean".equals(beanClassName) || "org.springframework.beans.factory.FactoryBean".equals(beanClassName));
+    }
+
+    @Override
+    public boolean isService(Object bean, String beanName) {
+        String beanClassName = bean.getClass().getName();
+        return isHsf && "com.taobao.hsf.app.spring.util.HSFSpringProviderBean".equals(beanClassName);
+    }
+
+    @Override
+    public RemotingDesc getServiceDesc(Object bean, String beanName) throws FrameworkException {
+        if (!this.isRemoting(bean, beanName)) {
+            return null;
+        }
+        try {
+            if (isReference(bean, beanName)) {
+                Object consumerBean = ReflectionUtil.getFieldValue(bean, "consumerBean");
+                Object metadata = ReflectionUtil.invokeMethod(consumerBean, "getMetadata");
+
+                Class<?> interfaceClass = (Class<?>) ReflectionUtil.invokeMethod(metadata, "getIfClazz");
+                String interfaceClassName = (String) ReflectionUtil.invokeMethod(metadata, "getInterfaceName");
+                String uniqueId = (String) ReflectionUtil.invokeMethod(metadata, "getVersion");
+                String group = (String) ReflectionUtil.invokeMethod(metadata, "getGroup");
+
+                RemotingDesc serviceBeanDesc = new RemotingDesc();
+                serviceBeanDesc.setInterfaceClass(interfaceClass);
+                serviceBeanDesc.setInterfaceClassName(interfaceClassName);
+                serviceBeanDesc.setUniqueId(uniqueId);
+                serviceBeanDesc.setGroup(group);
+                serviceBeanDesc.setProtocol(Protocols.HSF);
+                return serviceBeanDesc;
+            } else if (isService(bean, beanName)) {
+                Object consumerBean = ReflectionUtil.getFieldValue(bean, "providerBean");
+                Object metadata = ReflectionUtil.invokeMethod(consumerBean, "getMetadata");
+
+                String interfaceClassName = (String) ReflectionUtil.invokeMethod(metadata, "getInterfaceName");
+                Class<?> interfaceClass = (Class<?>) Class.forName(interfaceClassName);
+                String uniqueId = (String) ReflectionUtil.invokeMethod(metadata, "getVersion");
+                String group = (String) ReflectionUtil.invokeMethod(metadata, "getGroup");
+                RemotingDesc serviceBeanDesc = new RemotingDesc();
+                serviceBeanDesc.setInterfaceClass(interfaceClass);
+                serviceBeanDesc.setInterfaceClassName(interfaceClassName);
+                serviceBeanDesc.setUniqueId(uniqueId);
+                serviceBeanDesc.setGroup(group);
+
+                Object targetBean = ReflectionUtil.getFieldValue(metadata, "target");
+                serviceBeanDesc.setTargetBean(targetBean);
+                serviceBeanDesc.setProtocol(Protocols.HSF);
+                return serviceBeanDesc;
+            }
+        } catch (Throwable t) {
+            throw new FrameworkException(t);
+        }
+        return null;
+    }
+
+
+    @Override
+    public short getProtocol() {
+        return Protocols.HSF;
+    }
+}

--- a/tcc/src/main/resources/META-INF/services/io.seata.rm.tcc.remoting.RemotingParser
+++ b/tcc/src/main/resources/META-INF/services/io.seata.rm.tcc.remoting.RemotingParser
@@ -1,3 +1,4 @@
 io.seata.rm.tcc.remoting.parser.DubboRemotingParser
 io.seata.rm.tcc.remoting.parser.LocalTCCRemotingParser
 io.seata.rm.tcc.remoting.parser.SofaRpcRemotingParser
+io.seata.rm.tcc.remoting.parser.HSFRemotingParser

